ledger/catchupaccessor.go

```diff
@@ -690,18 +691,22 @@ func (c *catchpointCatchupAccessorImpl) BuildMerkleTrie(ctx context.Context, pro
 					return
 				}
 				trie.SetCommitter(mc)
-				for _, accountHash := range hashesToWrite {
+				for _, hash := range hashesToWrite {
 					var added bool
-					added, err = trie.Add(accountHash)
+					added, err = trie.Add(hash)
 					if !added {
-						return fmt.Errorf("CatchpointCatchupAccessorImpl::BuildMerkleTrie: The provided catchpoint file contained the same account more than once. hash '%s'", hex.EncodeToString(accountHash))
+						return fmt.Errorf("CatchpointCatchupAccessorImpl::BuildMerkleTrie: The provided catchpoint file contained the same account more than once. hash '%s'", hex.EncodeToString(hash))
 					}
 					if err != nil {
 						return
 					}
+
 				}
 				uncommitedHashesCount += len(hashesToWrite)
-				hashesWritten += uint64(len(hashesToWrite))
+				if progressUpdates != nil {
+					progressUpdates(hashesToWrite)
+				}
+
```

Comment:
Why the change to call the callback more often?

Reply:
@jannotti I'm open to feedback here - I thought it'd be least error prone to:
* Consolidate the hash identification logic in the listener rather than inline here.
* Consequently, the listener receives `[][]byte` rather than `uint64`.  And so I felt it'd be most robust to _avoid_ exposing `hashesWritten` beyond the immediate for-loop.

Reply:
I see, yes, makes sense.